frontend/src/Frontend/UI/Dialogs/SigBuilder.hs

```diff
@@ -0,0 +1,735 @@
+        text $ showWithDecimal kdaAmount <> " KDA"
+    if tokens' == mempty then blank else
+      void $ mkLabeledClsInput True "Amount (Tokens)" $ const $ el "div" $
+        forM_ (Map.toList tokens') $ \(name, amount) ->
+          el "p" $ text $ showWithDecimal amount <> " " <> name
+    void $ mkLabeledClsInput True "Unscoped Sigs" $
+      const $ text $ tshow $ _ts_numUnscoped summary
+  where
+    prpc (Exec _) = "Exec"
+    prpc (Continuation _) = "Continuation"
+
+
+--------------------------------------------------------------------------------
+-- Signature and Submission
+--------------------------------------------------------------------------------
+data SigDetails = SigDetails_Yaml | SigDetails_Json
+  deriving (Eq,Ord,Show,Read,Enum,Bounded)
+
+showSigDetailsTabName :: SigDetails -> Text
+showSigDetailsTabName SigDetails_Json = "JSON"
+showSigDetailsTabName SigDetails_Yaml = "YAML"
+
+signatureDetails
+  :: (MonadWidget t m)
+  => SigData Text
+  -> m ()
+signatureDetails sd = do
+  void $ mkLabeledInput False "Hash"
+    (\c -> uiInputElement $ c & initialAttributes %~ Map.insert "disabled" "") $ def
+      & inputElementConfig_initialValue .~ hashToText (toUntypedHash $ _sigDataHash sd)
+
+  divClass "tabset" $ mdo
+    curSelection <- holdDyn SigDetails_Yaml onTabClick
+    (TabBar onTabClick) <- makeTabBar $ TabBarCfg
+      { _tabBarCfg_tabs = [minBound .. maxBound]
+      , _tabBarCfg_mkLabel = const $ text . showSigDetailsTabName
+      , _tabBarCfg_selectedTab = Just <$> curSelection
+      , _tabBarCfg_classes = mempty
+      , _tabBarCfg_type = TabBarType_Primary
+      }
+
+    tabPane mempty curSelection SigDetails_Yaml $ do
+      let sigDataText = T.decodeUtf8 $ Y.encode1Strict sd
+      void $ uiSignatureResult sigDataText
+    tabPane mempty curSelection SigDetails_Json $ do
+      let sigDataText = T.decodeUtf8 $ LB.toStrict $ A.encode $ A.toJSON sd
+      void $ uiSignatureResult sigDataText
+  pure ()
+  where
+    uiSignatureResult txt = do
+      void $ uiTextAreaElement $ def
+        & textAreaElementConfig_initialValue .~ txt
+        & initialAttributes <>~ fold
+          [ "disabled" =: "true"
+          , "class" =: " labeled-input__input labeled-input__sig-builder"
+          ]
+      uiDetailsCopyButton $ constant txt
+
+signAndShowSigDialog
+  :: SigBuilderWorkflow t m model key
+  => model
+  -> SigBuilderRequest key
+  -> [KeyPair key]  -- ChainweaverKeys
+  -> Workflow t m (Event t ()) -- Workflow for going back
+  -> [(PublicKeyHex, UserSig)] -- User-supplied Sigs
+  -> Workflow t m (Event t ())
+signAndShowSigDialog model sbr keys backW sigsOrPrivate = Workflow $ mdo
+  onClose <- modalHeader $ text "Sig Data"
+  -- This allows a "loading" page to render before we attempt to do the really computationally
+  -- expensive sigs
+  pb <- delay 0.1 =<< getPostBuild
+  -- TODO: Can we forkIO the sig process and display something after they are done?
```

Comment:
So the idea was still to fork after the user presses "Accept / Sign". But i wanted finer grain changes so we could present a better loading screen like: " Loading: 3/8 signatures added" -- instead of just being stuck on it as it adds all 8 sigs at once